app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt

```diff
@@ -21,13 +21,10 @@ class Example1Activity : BaseMvvmActivity<Example1ViewModel>() {
         return Example1ViewModel(configuration)
     }
 
-    override fun onStart() {
-        super.onStart()
-        onStopDisposable += viewModel.eventPublisher.subscribe {
-            when (it) {
-                is Example1ViewEvent.GotoExample2 -> gotoExample2()
-                is Example1ViewEvent.ShowCurrentTimeToast -> showToast(it.message)
-            }
+    override fun onViewEvent(event: ViewEvent) {
+        when (event) {
```

Comment:
What do you think about making this when statement an exhaustive expression?

Reply:
with the current setup, it would take some extra boilerplate to do some class check/cast since ViewEvent would contain all events from other screens as well.

Reply:
For just dealing with side-effects, it is hard to do an exhaustive `when`. The `when` needs to be an expression if the compiler were to be able to check its exhaustiveness. In this case, the `when` is just dealing with side-effects; it is returning no value. 

If it is important to have this `when` check and properly handle *all* `Example1ViewEvent`, the `eventPublisher` should be declared to only emit values of that type (not of its super-type `ViewEvent`). And the `when` should return something. Assume that `it` is always of type `Example1ViewEvent` in the below example:

```
val action : () -> Unit = when (it) {
    is Example1ViewEvent.GotoExample2 -> ::gotoExample2
    is Example1ViewEvent.ShowCurrentTimeToast -> { { showToast(it.message) } }
}
action()
```

Reply:
What about something like this? https://proandroiddev.com/til-when-is-when-exhaustive-31d69f630a8b

Reply:
I'm not in favor of that. If *all* branches *must* produce a side-effect, an action, then the `when` expression must return an action (in the form of a lambda)

Reply:
I'm not in favor of that. If *all* branches *must* produce a side-effect, an action, then the `when` expression must return an action (in the form of a lambda)

Reply:
Ok - I'm fine with backing down on this one.  Thanks for the input!